### PR TITLE
fix: Disables `GenericMenu` without any sections or items

### LINKED ISSUE #30159: Disabling 'access-marketplace' permission for users show an empty 'Administration' tab
### Description:

When the 'access-marketplace' permission is disabled for users, the 'Administration' menu is empty (except for a vertical bar).

### Steps to reproduce:

1. Disable 'access-marketplace' permission for users
2. Log in as a normal user
3. Click on the 'Administration' menu

### Expected behavior:

Maybe simply hide the 'administration' button if the menu is empty? Or a least show a message such as 'No available action'.

### Actual behavior:

Weird empty menu (vertical bar showing)

image

### Server Setup Information:

- Version of Rocket.Chat Server: 6.3.1
- Deployment Method: docker
- MongoDB Version: 5.0

### Client Setup Information

Browser: Firefox

### LINKED ISSUE #30312: Disabling "create new" permissions for users shows an empty bar
### Description:

When the admin disables ''Create Private Channels", "Create Public Channels", "Create Team", "Start Discussion", "Start Discussion (Other-User)" and "Create Direct Messages" permissions for the users, the 'Create new' menu is empty and a bar is shown when we click on it.

### Steps to reproduce:

1.Disable ''Create Private Channels", "Create Public Channels", "Create Team", "Start Discussion", "Start Discussion (Other-User)" and "Create Direct Messages" permissions for users
2.Log in as a normal user
3.Click on the "Create new" menu

### Expected behavior:
Button should be disabled

### Actual behavior:

image

### Server Setup Information:

-Version of Rocket.Chat Server: 6.3.1
-Deployment Method: docker
-MongoDB Version: 5.0

### Client Setup Information

- Desktop App or Browser Version: Firexox
- Operating System: Ubuntu 22.04.2 LTS x86_64

### PR DESCRIPTION
**Pull request by anjy7**
## Proposed changes (including videos or screenshots)

Screenshot from 2023-09-07 01-51-22
The "Create new" button is disabled as it should be when the user doesn't have those permissions

## Issue(s)

Fixes #30312 
Fixes #30159
## Steps to test or reproduce

1.Disable ''Create Private Channels", "Create Public Channels", "Create Team", "Start Discussion", "Start Discussion (Other-User)" and "Create Direct Messages" permissions for users
2.Log in as a normal user
3.Click on the "Create new" menu

### CODE REVIEW

**Comment by anjy7** [comment]
Fixes #30159 as well

**Comment by anjy7** [comment]
@dougfabris could you please have a look at this

File: apps/meteor/client/components/GenericMenu/GenericMenu.tsx
Comment (dougfabris) [inline apps/meteor/client/components/GenericMenu/GenericMenu.tsx:1]: The `GenericMenu` should render conditionally an `IconButton` disabled when there are no sections and no items as well

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/client/components/GenericMenu/GenericMenu.tsx
Comment (anjy7) [reply apps/meteor/client/components/GenericMenu/GenericMenu.tsx:1]: Yup, done

**Comment by anjy7** [comment]
Hey @dougfabris,  I have made the requested changes. Please have a look

**Review by dougfabris — CHANGES_REQUESTED** [review]

**Comment by anjy7** [comment]
> @anjy7 Can you sign the CLA, please?

done, thank you

**Pull request merged**